### PR TITLE
Output classes and not class indexes as predictions

### PR DESCRIPTION
**Pull request by Ayaro**

### CODE REVIEW

**Comment by Ayaro** [comment]
@mfeurer Different implementation than [sklearn](https://github.com/scikit-learn/scikit-learn/blob/c957249/sklearn/ensemble/forest.py#L478), but allows to use different label types. Do we need it?
